app/controllers/admin/principals_controller.rb

```diff
@@ -28,8 +28,10 @@ def destroy
 
     message = destroy_message(principal)
 
-    user.principal.destroy
-    user.destroy
+    ActiveRecord::Base.transaction do
+      user.principal.destroy!
+      user.destroy!
+    end
```

Comment:
Nice!

Reply:
Is there any reason why we cannot let Rails take care of the ref integrity as it's done here:
https://github.com/moneyadviceservice/rad/blob/master/app/models/principal.rb#L11

And just call `principal.destroy`?